### PR TITLE
task 60 solved

### PR DESCRIPTION
**Pull request by arthur-borisenko**

### CODE REVIEW

File: week3/60-kvadratnoie-uravnieniie-2/task.py [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+
+
+class rootCounts:
+    NaN = "3"
+    Two = "2"
+    One = "1"
+    Zero = "0"
+
+
+def if_n_is_integer_convert_to_int(n):
```
Comment (IgorNB) [inline week3/60-kvadratnoie-uravnieniie-2/task.py:11]: safe_to_int?

File: week3/60-kvadratnoie-uravnieniie-2/task.py [hunk 2]
```diff
@@ -0,0 +1,52 @@
+import math
+
+
+class rootCounts:
```
Comment (IgorNB) [inline week3/60-kvadratnoie-uravnieniie-2/task.py:4]: is this Enum? then class RootCounts(Enum)?

class name must have UpperCase first letter

File: week3/60-kvadratnoie-uravnieniie-2/task.py [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+    elif b == 0:
+        return rootCounts.Zero
+    else:
+        return rootCounts.One, str(
+            if_n_is_integer_convert_to_int((-c) / b))
+
+
+def main():
+    a, b, c = float(input()), float(input()), float(input())
+    if a == 0:
```
Comment (IgorNB) [inline week3/60-kvadratnoie-uravnieniie-2/task.py:45]: make REUSABLE method for solving quadratic equations that have FULL logic to solve equation. Do not put part of solving logic in main

File: week3/60-kvadratnoie-uravnieniie-2/task.py [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+
+
+def if_n_is_integer_convert_to_int(n):
+    if n.is_integer():
+        return int(n)
+    else:
+        return n
+
+
+def calculate_roots(a, b, c):
```
Comment (IgorNB) [inline week3/60-kvadratnoie-uravnieniie-2/task.py:18]: solve_quadratic_equation?

File: week3/60-kvadratnoie-uravnieniie-2/task.py [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+    D = b ** 2 - 4 * a * c
+    if D == 0:
+        x = if_n_is_integer_convert_to_int(-b / (2 * a))
+        return rootCounts.One, str(x)
```
Comment (IgorNB) [inline week3/60-kvadratnoie-uravnieniie-2/task.py:22]: str(x) is part of printing results - not part of solving equation. try to return numbers from ewquation solver. try to format result for output independently

**Review by IgorNB — CHANGES_REQUESTED** [review]

File: week3/60-kvadratnoie-uravnieniie-2/task.py (hunk 2 above)
Comment (arthur-borisenko) [reply week3/60-kvadratnoie-uravnieniie-2/task.py]: done

File: week3/60-kvadratnoie-uravnieniie-2/task.py (hunk 3 above)
Comment (arthur-borisenko) [reply week3/60-kvadratnoie-uravnieniie-2/task.py]: done

File: week3/60-kvadratnoie-uravnieniie-2/task.py (hunk 4 above)
Comment (arthur-borisenko) [reply week3/60-kvadratnoie-uravnieniie-2/task.py]: done

File: week3/60-kvadratnoie-uravnieniie-2/task.py (hunk 5 above)
Comment (arthur-borisenko) [reply week3/60-kvadratnoie-uravnieniie-2/task.py]: done

**Review by IgorNB — CHANGES_REQUESTED** [review]

File: week3/60-kvadratnoie-uravnieniie-2/task.py [hunk 6]
```diff
@@ -0,0 +1,51 @@
+import math
+from enum import Enum
+
+
+class RootCounts(Enum):
+    NaN = 3
```
[the remark this answers is not in the document]
Comment (arthur-borisenko) [reply week3/60-kvadratnoie-uravnieniie-2/task.py:6]: done

File: week3/60-kvadratnoie-uravnieniie-2/task.py [hunk 7]
```diff
@@ -0,0 +1,51 @@
...
+    if hasattr(n, "is_integer") and n.is_integer():
+        return int(n)
+    else:
+        return n
+
+
+def safe_solve_quadratic_equation(a, b, c):
+    if a == 0:
+        if b == 0 and c == 0:
+            return [RootCounts.NaN.value]
```
[the remark this answers is not in the document]
Comment (arthur-borisenko) [reply week3/60-kvadratnoie-uravnieniie-2/task.py]: done

**Review by IgorNB — APPROVED** [review]

**Pull request merged**